tests/integration/utils.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+import tenacity
+import yaml
+from lightkube import ApiError
+from lightkube.generic_resource import create_namespaced_resource
+
+LOGGER = logging.getLogger(__name__)
+
+EXPERIMENT = create_namespaced_resource(
+    group="kubeflow.org",
+    version="v1beta1",
+    kind="experiment",
+    plural="experiments",
+    verbs=None,
+)
+
+TRIAL = create_namespaced_resource(
+    group="kubeflow.org",
+    version="v1beta1",
+    kind="trial",
+    plural="trials",
+    verbs=None,
+)
+
+
+def create_experiment(client, exp_path, namespace) -> str:
+    """Create Experiment instance."""
+    with open(exp_path) as f:
+        exp_yaml = yaml.safe_load(f.read())
+    exp_object = EXPERIMENT(exp_yaml)
+    client.create(exp_object, namespace=namespace)
+    return exp_yaml["metadata"]["name"]
+
+
+def delete_experiment(client, name, namespace):
+    """Delete Experiment instance."""
+    client.delete(EXPERIMENT, name=name, namespace=namespace)
+
+
+@tenacity.retry(
+    wait=tenacity.wait_exponential(multiplier=1, min=1, max=15),
+    stop=tenacity.stop_after_delay(30),
+    reraise=True,
+)
+def assert_experiment_exists(client, name, namespace):
+    """Asserts on the presence of the experiment in the cluster.
+
+    Retries multiple times using tenacity to allow time for the experiment
+    to be created.
+    """
+    try:
+        client.get(EXPERIMENT, name=name, namespace=namespace)
+    except ApiError:
+        raise AssertionError(
+            f"Waited too long to get experiment. \
+            Experiment {name} in namespace {namespace} does not exist when it should."
+        )
```

Comment:
Shouldn't we check the error status code? We can't really be certain that this is always gonna be a 404, can we?

Reply:
made a change there